src/plugins/index_pattern_management/public/components/index_pattern_table/index_pattern_table.tsx

```diff
@@ -135,13 +135,7 @@ export const IndexPatternTable = ({ canSave, history }: Props) => {
       setCreationOptions(options);
       setIndexPatterns(gettedIndexPatterns);
     })();
-  }, [
-    history.push,
-    indexPatterns.length,
-    indexPatternManagementStart,
-    uiSettings,
-    savedObjects.client,
-  ]);
+  }, [history.push, indexPatternManagementStart, uiSettings, savedObjects.client]);
```

Comment:
Good catch.

Reply:
yes, due to the fact that useEffect had variable indexPatterns.length, in its dependencies, useEffect worked 3 times with 3 requests, one of which was a cross-cluster and quite heavy. This fixes doing useEffect more than once :)